### ISSUE TITLE
Add support for using `bdist_nuitka` without setup.py, i.e. pure pyproject setup

### ISSUE THREAD

**Issue by nschloe** [Nuitka/Nuitka#1285]
One can build Nuitka'd wheels via
```
python3 setup.py bdist_nuitka
```
Is it possible to do the same in a [PEP 517](https://www.python.org/dev/peps/pep-0517/)/[518 setting](https://www.python.org/dev/peps/pep-0518/), i.e., without a `setup.py`, but with a `setup.cfg` and a `pyproject.toml`,
```toml
[build-system]
requires = ["setuptools>=42", "wheel"]
build-backend = "setuptools.build_meta"
```
?

**Comment by kayhayen** [comment]
Since it still uses `setuptools` and is only supposed to be a means to allow replacing it, it's not too relevant, I will add a test case for this now and check if our hook gets a chance to run. But it seems `setuptools.build_meta` just runs through normal setiup.py targets. So you would have to specifiy `build_with_nuitka=True` in the setup configuration. We do not have tests of that, because we use `bdist_wheel` and `bdist_nuitka` for easy switching without changing files. Also we do not use `setup.cfg` and it seems with `setuptools>=48` it has become independent of it maybe.

**Comment by kayhayen** [comment]
Added a `setup.cfg` using test case, it works of course. Now adding one with pyproject file and not setup.py which ought to work with newst setuptools.

**Comment by nschloe** [comment]
> So you would have to specifiy build_with_nuitka=True in the setup configuration.

Yeah, that's what I'm doing right now. I just have a stub `setup.py`,
```python
from setuptools import setup

if __name__ == "__main__":
    setup(build_with_nuitka=True)
```

> We do not have tests of that, because we use bdist_wheel and bdist_nuitka for easy switching without changing files.

Do you mean `python setup.py bdist_wheel` vs. `python setup.py bdist_nuitka`? This is what they're trying to phase out, see https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html.

> Also we do not use setup.cfg and it seems with setuptools>=48 it has become independent of it maybe.

`setup.cfg` is a better alternative for `setup.py`, and the best available option yet. With [PEP 621](https://www.python.org/dev/peps/pep-0621/), they're trying to move to a fully `pyproject.toml`-based installation, but `setup.cfg` isn't going to be deprecated soon.

**Comment by kayhayen** [comment]
I didn't manage to specify this option in a `setup.cfg` yet, is that not possible?

**Comment by nschloe** [comment]
Not sure here. Perhaps @pganssle can say something about if/how to make Nuitka hook into the `pyproject.toml`/`setup.cfg` packaging.

**Comment by kayhayen** [comment]
It seems Nuitka targets does not even get loaded. I am not the original author of that integration, but I believe I saw it working before, it seems `distutils.setup_keywords` is not called anymore.

**Comment by nschloe** [comment]
[`distutils` is being deprecated altogether](https://www.python.org/dev/peps/pep-0632/) so one should probably not rely too much on it anymore.

**Comment by kayhayen** [comment]
Deprecations do not matter much to Nuitka. People will use what they have always used. I managed to actually see the entry point being listed, and with a change in setup.py it actually work. Now I just need to figure out what format it takes for setup.cfg to also trigger it.

**Comment by kayhayen** [comment]
So this is working: 
```
[metadata]
name = nuitka-example-1
version = 101
description = yadda yadda.
long_description = Really long text here.
keywords = one, two
author = Nobody really
author_email = email@someplace.com
build_with_nuitka = True
```

Now need to see if the pyproject file passes that on too.

**Comment by kayhayen** [comment]
It's not, OK.. need to figure out, if and how the `setuptools.build_meta` does allow hooking.

**Comment by nschloe** [comment]
> Deprecations do not matter much to Nuitka. People will use what they have always used. 

They won't after Python 3.12. :smile: 
```
>>> import distutils
<stdin>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
```

**Comment by kayhayen** [comment]
Pardon my ignorance, I call `setuptools` always `distutils`. I lost track of the forking and replacing processes a long time ago there. It still mentions easy install.

However, looking at `setuptools.meta` I have come to believe that using a `nuitka.build_meta` that just wraps it and changes only `bdist_wheel` into `bdist_nuitka` might be trivial. I wonder if monkey patching it would be OK too.

**Comment by kayhayen** [comment]
Yup, seems to work, less than 20 lines of code. I will write something for the User Manual as part of my roadmap action for the standard thing, but basically you need to put this: 

```
requires = ["setuptools>=42", "wheel", "nuitka"]
build-backend = "nuitka.distutils.Build"
```

The build backend will be part of 0.6.18 likely. I need to figure out test integration still.

**Comment by kayhayen** [comment]
This is technically solved, but still underdocumented, I will add something as part of the Wheels roadmap update, and share it onTwitter.

**Comment by kayhayen** [comment]
This is currently on https://nuitka.net/doc/factory.html

**Comment by kayhayen** [comment]
This is currently on develop branch and will be released soon.

**Comment by kayhayen** [comment]
This is part of the 0.6.18 release and even documented somewhat in the User Manual. Let me know if you think these docs are sufficient.

**Comment by bsvedin** [comment]
This is exactly what I need. Thank you for implementing it.
Unfortunately, I am unable to get it to work. I made sure that I have the latest version of Nuitka - 0.6.18.1
I also have the latest version of poetry 1.1.12 - I also tried the preview version 1.2.0a2
When running "poetry build -f wheel" it still builds open source wheel files.
I tried having nuitka installed in my global site packages and in the poetry venv. Nuitka is also part of my path (C:\python36\Scripts) and I can call nuitka directly.
I even tried adding a print("BLAH BLAH") to the nuitka.distutils.Build file (both global and venv) and that doesn't get printed.
